### PR TITLE
Fix INTEGRATION_camera/shader_selection on macOS

### PR DESCRIPTION
**Pull request by mjcarroll**
# 🦟 Bug fix

Partially addresses #695

## Summary

This was a test that was disabled on macOS after metal porting, but now it can be re-enabled.

## Checklist
- [ ] Signed all commits for DCO
- [ ] Added tests
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if the library has them)
- [ ] `codecheck` passed (See [contributing](https://gazebosim.org/docs/all/contributing#contributing-code))
- [ ] All tests passed (See [test coverage](https://gazebosim.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Agazebosim+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

File: test/integration/camera.cc [hunk 1]
```diff
@@ -534,12 +534,27 @@ TEST_F(CameraTest, GZ_UTILS_TEST_ENABLED_ONLY_ON_LINUX(ShaderSelection))
...
+    switch(this->engine->GraphicsAPI())
+    {
+      case GraphicsAPI::OPENGL:
+      case GraphicsAPI::VULKAN:
+        vertexShaderFile = "simple_color_330_vs.glsl";
+        fragmentShaderFile = "simple_color_330_fs.glsl";
+        break;
+      case GraphicsAPI::METAL:
+        vertexShaderFile = "simple_color_vs.metal";
+        fragmentShaderFile = "simple_color_fs.metal";
```
Comment (chapulina) [inline test/integration/camera.cc:549]: It's still failing on mac

```
17: [ RUN      ] CameraTest.ShaderSelection
17: [Err] [Ogre2Material.cc:1382] Vertex shader path does not exist: /Users/jenkins/workspace/ignition_rendering-ci-pr_any-homebrew-amd64/ign-rendering/test/media/materials/programs/simple_color_vs.metal
17: [Err] [Ogre2Material.cc:1475] Fragment shader path does not exist: /Users/jenkins/workspace/ignition_rendering-ci-pr_any-homebrew-amd64/ign-rendering/test/media/materials/programs/simple_color_fs.metal
 17/210 Test  #17: INTEGRATION_camera_ogre2_metal .......................***Exception: SegFault  0.88 sec
```

**Comment by iche033** [comment]
I added the metal shaders in [d8bf582](https://github.com/gazebosim/gz-rendering/pull/703/commits/d8bf58218a9799c3cf8ada05348a850a2f7ffaa5) which fixed the test for me locally.

**Comment by mjcarroll** [comment]
> I added the metal shaders in [d8bf582](https://github.com/gazebosim/gz-rendering/pull/703/commits/d8bf58218a9799c3cf8ada05348a850a2f7ffaa5) which fixed the test for me locally.

D'oh.  I somehow missed committing these

**Comment by mjcarroll** [comment]
There is a known failing test on Windows that is addressed elsewhere, I'm going to merge.

**Comment by scpeters** [comment]
this test appears to pass in CI on [omnidroid](https://build.osrfoundation.org/computer/mac-omnidroid.bigsur) ([Build Status](https://build.osrfoundation.org/job/ignition_rendering-ci-pr_any-homebrew-amd64/2720/)), but it fails on t1000 and ultron (see results of https://build.osrfoundation.org/view/ign-garden/job/ignition_rendering-ci-gz-rendering7-homebrew-amd64/)

**Comment by scpeters** [comment]
from limited testing on `t1000`, I see the following at the end of `~/.gz/rendering/ogre2.log` after running the failing test:

~~~
12:13:23: Added resource location '/usr/local/Cellar/gz-rendering7/HEAD/share/gz/gz-rendering7/media/materials/textures/' of type 'FileSystem' to resource group 'General'
12:13:23: Added resource location '/Users/jenkins/workspace/ignition_rendering-ci-gz-rendering7-homebrew-amd64/ign-rendering/test/media/materials/programs/simple_color_vs.metal' of type 'FileSystem' to resource group 'General'
12:13:23: Shader _gz_simple_color_vs.metal compiled successfully.
12:13:23: Shader _gz_scene::Material(65526)_solid_fs compiled successfully.
12:13:23: Added resource location '/Users/jenkins/workspace/ignition_rendering-ci-gz-rendering7-homebrew-amd64/ign-rendering/test/media/materials/programs/simple_color_fs.metal' of type 'FileSystem' to resource group 'General'
12:13:23: Shader _gz_simple_color_fs.metal compiled successfully.
12:13:23: Can't assign material scene::Material(65523) because this Material does not exist. Have you forgotten to define it in a .material script?
12:13:23: Added resource location '/Users/jenkins/workspace/ignition_rendering-ci-gz-rendering7-homebrew-amd64/ign-rendering/test/media/materials/programs/simple_color_vs.metal' of type 'FileSystem' to resource group 'General'
12:13:23: Shader _gz_simple_color_vs.metal compiled successfully.
12:13:23: Shader _gz_scene::Material(65521)_solid_fs compiled successfully.
12:13:23: Added resource location '/Users/jenkins/workspace/ignition_rendering-ci-gz-rendering7-homebrew-amd64/ign-rendering/test/media/materials/programs/simple_color_fs.metal' of type 'FileSystem' to resource group 'General'
12:13:23: Shader _gz_simple_color_fs.metal compiled successfully.
12:13:23: Texture memory budget exceeded. Stalling GPU.
12:13:23: Stalling was not enough. Freeing memory.
~~~

The log file ends there. The test passes on my local machine and has some of these same lines that look concerning, but it doesn't stop there; it continues a bit farther.

### Successful run on my local machine:

~~~
12:03:23: Added resource location '/Users/scpeters/ws/rendering/install/gz-rendering7/share/gz/gz-rendering7/media/materials/textures/' of type 'FileSystem' to resource group 'General'
12:03:23: Added resource location '/Users/scpeters/ws/rendering/src/gz-rendering/test/media/materials/programs/simple_color_vs.metal' of type 'FileSystem' to resource group 'General'
12:03:23: Shader _gz_simple_color_vs.metal compiled successfully.
12:03:23: Shader _gz_scene::Material(65526)_solid_fs compiled successfully.
12:03:23: Added resource location '/Users/scpeters/ws/rendering/src/gz-rendering/test/media/materials/programs/simple_color_fs.metal' of type 'FileSystem' to resource group 'General'
12:03:23: Shader _gz_simple_color_fs.metal compiled successfully.
12:03:23: Can't assign material scene::Material(65523) because this Material does not exist. Have you forgotten to define it in a .material script?
12:03:23: Added resource location '/Users/scpeters/ws/rendering/src/gz-rendering/test/media/materials/programs/simple_color_vs.metal' of type 'FileSystem' to resource group 'General'
12:03:23: Shader _gz_simple_color_vs.metal compiled successfully.
12:03:23: Shader _gz_scene::Material(65521)_solid_fs compiled successfully.
12:03:23: Added resource location '/Users/scpeters/ws/rendering/src/gz-rendering/test/media/materials/programs/simple_color_fs.metal' of type 'FileSystem' to resource group 'General'
12:03:23: Shader _gz_simple_color_fs.metal compiled successfully.
12:03:23: Texture memory budget exceeded. Stalling GPU.
12:03:23: Stalling was not enough. Freeing memory.
12:03:24: Unregistering ResourceManager for type Font
12:03:24: DefaultWorkQueue('Root') shutting down on thread main.
12:03:24: *-*-* OGRE Shutdown
12:03:24: Unregistering ResourceManager for type OldSkeleton
12:03:24: Unregistering ResourceManager for type Mesh2
12:03:24: Unregistering ResourceManager for type Mesh
12:03:24: Unregistering ResourceManager for type Material
12:03:24: Unregistering ResourceManager for type HighLevelGpuProgram
12:03:24: Uninstalling plugin: Metal RenderSystem
12:03:24: Unregistering ResourceManager for type GpuProgram
12:03:24: Plugin successfully uninstalled
12:03:24: Unloading library /usr/local/opt/ogre2.3/lib/OGRE-2.3/OGRE/RenderSystem_Metal.dylib
12:03:24: Uninstalling plugin: ParticleFX
12:03:24: Plugin successfully uninstalled
12:03:24: Unloading library /usr/local/opt/ogre2.3/lib/OGRE-2.3/OGRE/Plugin_ParticleFX.dylib
12:03:24: Uninstalling plugin: GL 3+ RenderSystem
12:03:24: ***********************************************
***  Stopping Mac OS X OpenGL 3+ Subsystem  ***
***********************************************
12:03:24: Plugin successfully uninstalled
12:03:24: Unloading library /usr/local/opt/ogre2.3/lib/OGRE-2.3/OGRE/RenderSystem_GL3Plus.dylib
~~~

**Comment by iche033** [comment]
I also tested locally on my mac and couldn't reproduce the test failure. The `VisualAt` function is failing to give the right output. Other tests that use this function are already disabled on mac, e.g. the [VisualAt](https://github.com/gazebosim/gz-rendering/blob/gz-rendering7/test/integration/camera.cc#L165) integration test. We could also do the same here and revisit later.

**Comment by scpeters** [comment]
I added print statements down in the ogre2 selection buffer and material switcher, and it was not passing the right color into the material switcher color dictionary. I'm guessing the texture is not being loaded? Weird that it works on omnidroid but not others

**Comment by scpeters** [comment]
I'm attempting to prevent the test from seg-faulting during this failure case in https://github.com/gazebosim/gz-rendering/pull/720

This should improve the failure reporting.

**Comment by iche033** [comment]
> it was not passing the right color into the material switcher color dictionary. I'm guessing the texture is not being loaded?

or the texture is just incorrectly rendered. Needs some debugging

**Pull request merged**